pkg/kubelet/server/server.go

```diff
@@ -126,10 +128,16 @@ type containerInterface interface {
 // so we can ensure restful.FilterFunctions are used for all handlers
 type filteringContainer struct {
 	*restful.Container
+	oteltrace.TracerProvider
```

Comment:
We should be using the NoopTracerProvider instead of nil everywhere, so we don't have to handle nil checking.  I was planning to change that as a follow-up across both apiserver tracing and kubelet tracing.

Reply:
I updated to use NoopTracerProvider  in the last pushed commit, PTAL, thanks